packages/core/src/framework/augury-core.ts

```diff
@@ -1,53 +1,40 @@
 import { ChannelManager } from './channels';
-import { Command, CommandService } from './commands';
-import { EventDispatcher } from './dispatcher';
-import { Enhancer, EnhancerService } from './enhancers';
 import { HistoryManager } from './history';
 import { PluginManager } from './plugins';
 import { Plugin } from './plugins';
-import { Probe, ProbeManager } from './probes';
-import { Reaction, ReactionService } from './reactions';
+import { EventDispatcher, Probe } from './probes';
+import { Reducer } from './reducers';
+import { Scanner } from './scanner';
 
 export class AuguryCore {
-  private readonly dispatcher: EventDispatcher;
-  private readonly probeManager: ProbeManager;
-  private readonly enhancerService: EnhancerService;
-  private readonly channelManager: ChannelManager;
-  private readonly reactionService: ReactionService;
-  private readonly commandService: CommandService;
+  public readonly dispatcher: EventDispatcher;
+  public readonly channelManager: ChannelManager;
+  public readonly historyManager: HistoryManager;
   private readonly pluginManager: PluginManager;
-  private readonly historyManager: HistoryManager;
-
-  constructor(
-    probes: Probe[],
-    enhancers: Enhancer[],
-    reactions: Reaction[],
-    commands: Array<Command<any>>,
-    plugins: Plugin[],
-  ) {
-    this.probeManager = new ProbeManager(probes);
-    this.enhancerService = new EnhancerService(this.probeManager, enhancers);
+
+  constructor(probes: Probe[], plugins: Plugin[]) {
+    this.dispatcher = new EventDispatcher(probes);
     this.channelManager = new ChannelManager();
     this.historyManager = new HistoryManager();
-    this.reactionService = new ReactionService(
-      reactions,
-      this.probeManager,
-      this.channelManager,
-      this.historyManager,
-    );
-    this.dispatcher = new EventDispatcher(
-      this.probeManager,
-      this.enhancerService,
-      this.reactionService,
-      this.historyManager,
-    );
-    this.commandService = new CommandService(this.dispatcher, commands);
-    this.pluginManager = new PluginManager(this.commandService, plugins);
+    this.pluginManager = new PluginManager(plugins, this);
+
+    this.dispatcher.subscribe(event => {
+      event.markComplete();
+
+      this.historyManager.addEvent(event);
+    });
   }
 
   public initialize(ngZone, ngModule) {
-    this.probeManager.initialize(ngZone, ngModule);
+    this.dispatcher.initialize(ngZone, ngModule);
 
     return this;
   }
+
+  public createLiveChannel(reducer: Reducer) {
+    const scanner = new Scanner(reducer, this.historyManager);
+    scanner.scan(this.dispatcher);
+
+    return this.channelManager.createScannerChannel(scanner);
+  }
```

Comment:
I want to remove the scanner - this is just the identical code that was in the Reaction class. We'll get to it next pr.